### DISCUSSION TITLE
Consensus on suggestive names for the Veracruz components

### DISCUSSION THREAD

**Discussion by dominic-mulligan-arm** [veracruz-project/veracruz#83] [Ideas]
As mentioned in #33, there's a consensus that the Mexican names of the various Veracruz components should now be abandoned in favour of more descriptive component names.  Some suggestions (harvested from a previous set of suggestions by Derek) are listed in #33:

- Jalisco -> SGXRootEnclave
- Sonora -> TrustZoneRootEnclave
- Tabasco -> ProxyAttestationServer
- Colima -> TransportProtocol
- Durango -> VeracruzClient
- Sinaloa -> VeracruzServer
- Chihuahua -> ExecutionEngine
- Baja -> SessionManager
- MexicoCity -> RuntimeManager
- Tlaxcala -> WASMChecker

Does anybody have any qualms about us adopting these names, or can think of better ones?  Given the task will touch virtually everything in the Veracruz codebase, it's likely that any such renaming will be gradual.

**Comment by dreemkiller** [comment]
Sinaloa is going to be split per platform (per #81), as will Mexico City.

So, Sinaloa will eventually be split to TrustZoneVeracruzServer, SGXVeracruzServer, NitroVeracruzServer.

Mexico City will be split to TrustZoneRuntimeManager, SGXRuntimeManager, and NitroRuntimeManager.

**Comment by geky** [comment]
Just a note, Rust style suggests WasmChecker (as opposed to WASMChecker). And I suppose also SgxVeracruzServer and TrustzoneVeracruzServer.

https://rust-lang.github.io/api-guidelines/naming.html
> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

Though do we care? ¯\\\_(ツ)\_/¯ I don't have a strong opinion. It does avoid sometimes difficult to read names such as USBHID2TLSSpeedDriver.

**Reply by dreemkiller** [reply]
I think we should comply with community style standards unless we have strong reasons not to.

I certainly have some strong opinions that the Rust community disagrees with, but this ain't one of them.
